tensorflow_probability/python/distributions/pert.py

```diff
@@ -0,0 +1,164 @@
+"""The PERT distribution class"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow.compat.v2 as tf
+from tensorflow_probability.python.distributions import Beta
+from tensorflow_probability.python.bijectors import AffineScalar
+from tensorflow_probability.python.distributions import transformed_distribution
+from tensorflow_probability.python.internal import assert_util
+from tensorflow_probability.python.internal import dtype_util
+from tensorflow_probability.python.internal import prefer_static
+from tensorflow_probability.python.internal import tensor_util # pylint: disable=g-direct-tensorflow-import
+
+__all__ = [
+    'PERT',
+]
+
+class PERT(transformed_distribution.TransformedDistribution):
+  """Modified PERT distribution:
+  PERT distribution is a loc-scale family of Beta distribution intended to
+  model expert prediction on arbitrary real interval bounded by MAX and MIN
+  values set by the user, along with MODE to indicate the expert's
+  most frequent prediction.
+
+  The shift and scale factor is given by:
+
+  ```
+  shift = MIN
+  scale = MAX - MIN
+  ```
+
+  Modified PERT is a generalized version of PERT that has an additional
+  parameter, smoothness (g).
+
+  Smoothness determines the properties of Beta distribution:
+
+  ```
+  concentration1 = 1 + g * (MODE - MIN)/(MAX - MIN)
+  concentration0 = 1 + g * (MAX - MODE)/(MAX - MIN)
+  mean = (MIN + g * MODE + MAX)/(g + 2)
+  ```
+
+  PERT distribution is obtained when g = 4.
+
+  For general reference: https://en.wikipedia.org/wiki/PERT_distribution
+  """
+  def __init__(self,
+               mini,
+               mode,
+               maxi,
+               smoothness=4.,
+               validate_args=False,
+               allow_nan_stats=False,
+               name="PERT"):
+    parameters = dict(locals())
+    with tf.name_scope(name) as name:
+      dtype = dtype_util.common_dtype([mini, maxi, mode], tf.float32)
+      self._min = tensor_util.convert_nonref_to_tensor(
+          mini, name='mini', dtype=dtype)
+      self._mod = tensor_util.convert_nonref_to_tensor(
```

Comment:
should this be `_mode`?

Reply:
I guess this won't matter anymore since I agree on changing these names to `low`, `peak`, and `high` but I am just curious about how to approach this kind of case in terms of general software engineering / design principle.

So I avoided calling this `_mode` because it seemed to be an abstract method in `distribution` class - and I didn't want to set a property on a name where a class usually expects a method. This is how I ended up setting it to `_mod` and implemented `_mode` method to return this property. Is this approach reasonable or is there a general good/alternative practice when it comes to these kind of things?